pandas/io/parsers/readers.py

```diff
@@ -542,13 +542,9 @@ def _read(
     """Generic reader of line files."""
     # if we pass a date_parser and parse_dates=False, we should not parse the
     # dates GH#44366
-    if (
-        kwds.get("date_parser", None) is not None
-        and kwds.get("parse_dates", None) is None
-    ):
-        kwds["parse_dates"] = True
-    elif kwds.get("parse_dates", None) is None:
-        kwds["parse_dates"] = False
+    if kwds.get("parse_dates", None) is None:
```

Comment:
I prefer the old version, more readable imo

Reply:
Readability IMO is subjective. What might be readable to me might be unreadable to someone else.

Anyway, I would consider the following more readable:

```
if kwds.get("parse_dates", None) is None:
    if kwds.get("date_parser", None) is not None:
        kwds["parse_dates"] = True
    else:
        kwds["parse_dates"] = False
```

Would like to know your opinion on the above.

Reply:
Yep this is better